### ISSUE TITLE
Deal with Jetpack sites connected to a different wpcom account

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#5134]
I hope there are only a very few users who log in a .com account and then try to add a Jetpack site connected to another .com account. IMO, this can wait for after fluxc release.

# How this can happen

1. **Case 1**: User tries to add a new self hosted site from the Site Picker.
2. **Case 2**: User already have a self hosted site in the app and then tries to connect to .com account.

# How we can detect the problem:

**Case 1** 

User adds a self hosted site, we pull it, it's a jetpack site, we try to do a REST call with the default account - it fails with an unauthorized error.

**Case 2**

User logs in a .com account, we iterate over all the jetpack sites in the site store and try a .com REST call on each of them - it fails with an unauthorized error.

# Solutions

**Case 1**

3 solutions:

1. Prevent it: unauthorized error after adding a site -> "You can't add this site, blablabla..."
2. Show a warning and force the site as pure self hosted (force isJetpackInstalled/Connected to false): "Your site has been added, but some capabilities disabled because blablabla...", we also need a new field in the SiteModel, something like "forcedSelfHosted". Can be a real PITA to deal with those if the user logs out / logs in another account or connect Jetpack to its other account.
3. Do nothing specific during the site addition process. The user will see errors everytime they tries to do something "Unable to fetch posts / stats / etc...", totally unusable in the app but we can show a specific message to educate the user.

**Case 2**

3 solutions:

1. Prevent log in: unauthorized error on a Jetpack site after signing in -> "You can't sign in this account, because blablabla, you can remove your site XXX and try again".
2. Show a warning and force the erroneous sites as pure self hosted (same problems as above).
3. Do nothing specific. (same as above).

My favorite solution: keep it simple, prevent them to be added to the app or prevent sign in, we don't want people to use different accounts and we really don't want to deal with zombie sites (forcedSelfHosted that could be later transformed to jetpack sites).

**Comment by aforcier** [comment]
The simple solutions make sense to me too as a first pass.

We'll probably need to account for this in self-hosted site migration on upgrade.

**Comment by maxme** [comment]
> We'll probably need to account for this in self-hosted site migration on upgrade.

We migrate "pure" self hosted sites only and we expect Jetpack sites to be pulled from the wpcom account (after account migration). So "bastard" jetpack sites won't be migrated (and that's a good thing IMO).

**Comment by aforcier** [comment]
> We migrate "pure" self hosted sites only and we expect Jetpack sites to be pulled from the wpcom account (after account migration). So "bastard" jetpack sites won't be migrated (and that's a good thing IMO).

This doesn't seem to be the case.

In `develop`, I signed into a WP.com account which has one pure WP.com site (`fluxcexample123.wordpress.com`), and one Jetpack site (`do.wpmt.co./jp-1`). I then added a self-hosted site, `do.wpmt.co./jp-2`, which has Jetpack but is connected to a different account.

![migration-1-develop](https://cloud.githubusercontent.com/assets/9613966/22709484/eba2405a-ed47-11e6-914c-b91f9078d339.png)

(Of course, `do.wpmt.co/jp-1` is missing since we don't pull Jetpack sites over WP.com REST in `develop`.)

Then, migrate to current `fluxc-integration`:

![migration-2-fluxc-integration](https://cloud.githubusercontent.com/assets/9613966/22709506/0548a788-ed48-11e6-8296-92eb27a34b5b.png)

`do.wpmt.co/jp-2` is still there. And we're calling it a Jetpack site:

```java
mIsJetpack = true
mIsWPCom = false
mLoginUrl = "http://do.wpmt.co/jp-1/wp-login.php"
mSiteId = 122536325
```
